r/R/array.R

```diff
@@ -269,6 +263,8 @@ filter_rows <- function(x, i, keep_na = TRUE, ...) {
   nrows <- x$num_rows %||% x$length() # Depends on whether Array or Table-like
   if (inherits(i, "array_expression")) {
     # Evaluate it
+    # Switch this when https://issues.apache.org/jira/browse/ARROW-9380 is resolved
+    # i <- eval_array_expression(i)
```

Comment:
This can be done now

Reply:
I'm doing this now and checking the tests

Reply:
When uncommenting this I get this failure

```
══ Failed ══════════════════════════════════════════════════════════════════════
── 1. Error: filter() with %in% (@test-dplyr.R#149)  ───────────────────────────
'match' requires vector arguments
Backtrace:
  1. arrow:::expect_dplyr_equal(...) testthat/test-dplyr.R:149:2
  2. dplyr::filter(., dbl > 2, chr %in% c("d", "f"))
 10. dplyr::collect(.)
 14. arrow:::`[.RecordBatch`(...)
 15. arrow:::filter_rows(x, i, ...)
 16. arrow:::eval_array_expression(i)
 17. base::lapply(...)
 18. arrow:::FUN(X[[i]], ...)
 19. arrow:::eval_array_expression(a)
 21. <environment> %in% c("d", "f")

══ DONE ════════════════════════════════════════════════════════════════════════
```

Reply:
Yes, I know, I started working on this and ran into ARROW-9389. I can hack around this here for now, but I was leaning towards holding this until after 1.0 anyway. What do you think?

Reply:
Oh right. 

I'll add metafunctions for calling "match" and "isin"? That's pretty easy, I'm not sure how else to resolve it